<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ORWidget</class>
 <widget class="QWidget" name="ORWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>786</width>
    <height>554</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Arial</family>
    <pointsize>8</pointsize>
    <bold>true</bold>
   </font>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>ORWidget</string>
  </property>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>10</y>
     <width>631</width>
     <height>521</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>1</width>
     <height>1</height>
    </size>
   </property>
   <property name="cursor">
    <cursorShape>CrossCursor</cursorShape>
   </property>
   <property name="mouseTracking">
    <bool>true</bool>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>10</y>
     <width>81</width>
     <height>151</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QPushButton" name="readImageButton">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>读入图片</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="moveButton">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>移动</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
      <property name="autoExclusive">
       <bool>true</bool>
      </property>
      <attribute name="buttonGroup">
       <string notr="true">buttonGroup</string>
      </attribute>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="drawRectButton">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>画矩形</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="autoExclusive">
       <bool>true</bool>
      </property>
      <attribute name="buttonGroup">
       <string notr="true">buttonGroup</string>
      </attribute>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="drawButton">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="mouseTracking">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>画图</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
      <property name="autoExclusive">
       <bool>true</bool>
      </property>
      <attribute name="buttonGroup">
       <string notr="true">buttonGroup</string>
      </attribute>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="writeImageButton">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>写图片</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QTextBrowser" name="mouseInfo">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>170</y>
     <width>81</width>
     <height>91</height>
    </rect>
   </property>
   <property name="mouseTracking">
    <bool>false</bool>
   </property>
   <property name="html">
    <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Arial'; font-size:8pt; font-weight:700; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QPushButton" name="undoButton">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>270</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="text">
    <string>撤销</string>
   </property>
  </widget>
  <widget class="QPushButton" name="redoButton">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>300</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text">
    <string>反向撤销</string>
   </property>
  </widget>
  <widget class="QPushButton" name="setPenColorButton">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>330</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text">
    <string>设置颜色</string>
   </property>
  </widget>
  <widget class="QLabel" name="pen">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>370</y>
     <width>10</width>
     <height>10</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QSlider" name="verticalSlider">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>390</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <property name="maximum">
    <number>25</number>
   </property>
   <property name="pageStep">
    <number>3</number>
   </property>
   <property name="value">
    <number>5</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>writeImageButton</sender>
   <signal>clicked()</signal>
   <receiver>ORWidget</receiver>
   <slot>cv2Imwrite()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>702</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>51</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>readImageButton</sender>
   <signal>clicked()</signal>
   <receiver>ORWidget</receiver>
   <slot>cv2Imread()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>698</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>8</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>undoButton</sender>
   <signal>clicked()</signal>
   <receiver>ORWidget</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>290</y>
    </hint>
    <hint type="destinationlabel">
     <x>744</x>
     <y>354</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redoButton</sender>
   <signal>clicked()</signal>
   <receiver>ORWidget</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>692</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>759</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setPenColorButton</sender>
   <signal>clicked()</signal>
   <receiver>ORWidget</receiver>
   <slot>setPenColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>737</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>771</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>verticalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ORWidget</receiver>
   <slot>updatePen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>686</x>
     <y>460</y>
    </hint>
    <hint type="destinationlabel">
     <x>759</x>
     <y>460</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>cv2Imread()</slot>
  <slot>cv2Imwrite()</slot>
  <slot>painter()</slot>
  <slot>undo()</slot>
  <slot>redo()</slot>
  <slot>setPenColor()</slot>
  <slot>updatePen()</slot>
 </slots>
 <buttongroups>
  <buttongroup name="buttonGroup">
   <property name="exclusive">
    <bool>true</bool>
   </property>
  </buttongroup>
 </buttongroups>
</ui>
